### DISCUSSION TITLE
Support comments on columns (generate down to schema.rs)

### DISCUSSION THREAD

**Discussion by SamuelMarks** [diesel-rs/diesel#4375] [Ideas]
I like to use `struct`s to represent things all the way down to CLIs, OpenAPI [utoipa crate], and more.

My current `struct`s are uninformative:
```rs
#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, diesel::Queryable, diesel::Selectable, diesel::QueryableByName, PartialEq, diesel::Identifiable)]
#[diesel(table_name=my_model, primary_key(id))]
pub struct MyModel {
    /// Field representing column `id`
    pub id: i32,
}
```

(as generated via schema.rs with https://github.com/Wulf/dsync )

How should this be supported? - I'm thinking:
```sql
CREATE TABLE "my_model" (
    id             VARCHAR(50) PRIMARY KEY    -- user-modifiable id
);
```

Or maybe:
```sql
COMMENT ON COLUMN my_model.id IS 'user-modifiable id';
```

Or something else? - What do you think?

Happy to contribute a PR. Related: https://github.com/Wulf/dsync/issues/144

**Comment by weiznich** [comment]
Diesel already reads comment on tables and columns and emits them as doc comments in your schema.rs file. You need to configure diesel-cli to actually emit the doc comments, but otherwise this already works. See [this guide](https://diesel.rs/guides/configuring-diesel-cli.html) for details about the relevant setting.

**Reply by hasezoey** [reply]
Didnt know this existed, thanks for pointing this out. Maybe consider mentioning that `with-docs` (both in help and the mentioned guide) also reads and uses sql comments (and also if it is intended to work with sqlite?)
(and maybe also note that such a sql comment will overwrite the comment otherwise created by diesel?)

---

for anyone seeing this in the future, the output context is:

```rs
// @generated automatically by Diesel CLI.

diesel::table! {
    /// hello
    franchise (_id) {
        /// testy
        _id -> Integer,
    }
}
```

from (mariadb) sql:
```sql
-- test comment 1
CREATE TABLE franchise (
	-- test comment 2
	_id INTEGER NOT NULL PRIMARY KEY COMMENT "testy"
) comment = "hello";
```

**Reply by weiznich** [reply]
> Didnt know this existed, thanks for pointing this out. Maybe consider mentioning that with-docs (both in help and the mentioned guide) also reads and uses sql comments 

That sounds reasonable. I'm happy to accept a PR for this.

>  (and also if it is intended to work with sqlite?)

This doesn't work for sqlite as sqlite doesn't support the `COMMENT "your comment"` syntax, so it's not expected to work there.
